File: programmers/Python/Level2/프린터.py
def solution(priorities, location):
    answer = 0
    n = len(priorities)
    while priorities:
        max_pri = max(priorities)
        while priorities[0] != max_pri:
            priorities.append(priorities[0])
            priorities.pop(0)
            location = (location - 1) % n
        if location != 0:
            priorities.pop(0)
            n -= 1
            location -= 1
            answer += 1
        else:
            return answer + 1
    return -1

File: programmers/C++/Level2/프린터.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> priorities, int location) {

	int index = 0;
	for (int i = 1; i <= priorities.size(); i++)  // i : 출력 우선순위
	{
		int max_priority = *max_element(priorities.begin(), priorities.end());
		while (1)
		{
			if (priorities.at(index) == max_priority)
			{
				if (index == location) return i;
				priorities.at(index) = 0;
				break;
			}
			else index = (index + 1) % priorities.size();
		}
	}

	return -1;
}
